### ISSUE TITLE
UI scaling issues in iPadOS

### ISSUE THREAD

**Issue by FarawayMagnet** [hyvanmielenpelit/GnollHack#2089]
certain UI elements do not seem to scale properly on an ipad's screen. this is immediately visible in a brand-new game, during the introductory "Let's review the user interface" walkthrough. the red circles meant to call out various UI elements (buttons and such) do not align with the buttons they're meant to describe. see these screenshots for examples:

![IMG_0575](https://github.com/user-attachments/assets/66d2b0d2-63a4-49f5-b67f-169864fba4ae)
![IMG_0576](https://github.com/user-attachments/assets/77f2947a-a61c-4a30-a775-6054aa993232)

### device info:
iPad Air 11-inch (M2)

### app info:
GnollHack Version:4.2.0 (Build 15)
Configuration:Release
10S Port Version:1.82
iOS Port Build:138

**Comment by janne-hmp** [comment]
On most devices, including newest iPad Airs, the screen scaling for tips works normally ok, but we have observed that with some devices sometimes the scaling is not correct. Do you have any particular screen scaling settings? Also, does the incorrect scaling happen consistently every time if you launch the UI tips during the game from Game Menu > User Interface Guide?

**Comment by janne-hmp** [comment]
Has the newest version (1.85) still the same problem?

**Comment by janne-hmp** [comment]
Closed with no answer.

**Comment by FarawayMagnet** [comment]
@janne-hmp apologies, apparently my github email notifications had become disabled.

i just reinstalled the latest version, which appears to be 1.86, and the issue persists.

![IMG_0580](https://github.com/user-attachments/assets/57ac9c07-22f3-4b39-8006-b33f894c0e0f)

on my ipad, i have the "text size" setting (Settings app > Display & Brightness > Text Size) set to the furthest-left (smallest) option on the slider, but i have changed no other settings.

yes, this is consistently reproducible.

also, i've noticed another apparent scaling issue... there is some overlap between screen elements. see the inventory button indicated by my green arrow, top-right.

**Comment by TommiGustafsson-HMP** [comment]
Can you please post your iPad model and specs?

**Comment by janne-hmp** [comment]
Thanks, that should help us work on the issue.

**Comment by FarawayMagnet** [comment]
iPadOS Version    18.0
Model Name    iPad Air 11-inch (M2)
Model Number    MUWC3LL/A

**Comment by TommiGustafsson-HMP** [comment]
I tested on that iPad on the Browserstack App Live test cloud and it worked ok. It must be some setting on your part that's causing the problem.

**Comment by FarawayMagnet** [comment]
i pointed you to the wrong setting in an earlier comment. i don't think the "Text Size" setting is actually relevant, it's more likely to be the "Display Zoom" setting (Settings app > Display & Brightness > Display Zoom). this setting is set to "More Space".

**Comment by janne-hmp** [comment]
That's right, it is the Display Zoom - More Space, which causes things to be pointed at the wrong place. We'll have to check how to determine that it is on, and what scaling number it in fact changes. Currently, the game assumes that the logical pixels are scaled by DisplayDensity only, but this does not seem to be the case when Display Zoom is on (there will be an extra scaling factor).

**Comment by janne-hmp** [comment]
It looks like this has now been fixed.
